homeassistant/components/notion/__init__.py

```diff
@@ -181,6 +181,10 @@ def async_save_refresh_token(refresh_token: str) -> None:
     # Create a callback to save the refresh token when it changes:
     entry.async_on_unload(client.add_refresh_token_callback(async_save_refresh_token))
 
+    # Save the client's refresh token if it's different than what we already have:
+    if (token := client.refresh_token) and token != entry.data.get(CONF_REFRESH_TOKEN):
```

Comment:
A refresh token will always exist at this point, so this `.get` is a little loose. The tests need some renovation, so this will unstick this PR for now.